pt-BR/1-5.md

```diff
@@ -0,0 +1,86 @@
+
+Segundo, mesmo tendo certeza que o método `instantiateViewController()` irá nos responder com `DetailViewController`, Swift *acredita* que ele irá nos responder com um `UIViewController` porque a linguagem não consegue interpretar qual o tipo exato que foi descrito dentro do storyboard.
+
+Isto parecerá confuso se você é novo em programação, por isso, deixe-me tentar te explicar melhor usando uma analogia. Vamos dizer que você tem um encontro hoje a noite, diante disto você me pede para lhe conseguir dois ingressos para um evento. Sendo seu amigo eu saio e procuro os ingressos, coloco-os em um envelope e lhe entrego. Eu realizei a minha parte do combinado: Você pediu os ingressos, recebeu os ingressos. Mas que tipo de ingressos são? Ingressos para o teatro? Para um jogo de esportes? Para o metrô? A única maneira de você saber é abrir o envelope e conferir.
+
+A linguagem Swift tem o mesmo problema. O método `instantiateViewController()` foi definido como retornando um objeto do tipo `UIViewController`. Para a linguagem Swift, qualquer view controller que tenha sido criado usando como referência o padrão criado pela Apple é um `UIViewController`. Isto nos cria um problema porque o padrão criado pela Apple não possui a propriedade `selectedImage` que queremos definir. A solução: precisamos informar a linguagem que o que ela receber do método não é o que ela pensa que vai receber.
+
+O termo técnico para este procedimento em programação é "typecasting", o que se fosse traduzido para o português seria "conversão de tipo". Com este procedimento nós dizemos para qualquer linguagem de programação que se baseie em tipos para tratar determinado valor como sendo de um tipo diferente do tipo que ela espera. Com Swift temos diversas maneiras de fazer isto, por agora iremos usar a opção mais segura. E a segurança que ela nos proporciona é porque realizamos o nosso typecasting usando opcionais. O que faremos, se fosse traduzido para o português, seria mais ou menos o seguinte: "Tente converter o tipo que você recebrá para este outro tipo e caso seja bem sucedido na conversão use o valor. Caso não seja bem sucedido, não faça nada e siga em frente."
+
+Se conseguirmos obter um `UIViewController` convertido para um `DetailViewController` iremos definir a sua propriedade `selectedImage` com o valor que existe em `picures[indexPath.row]`, da mesma maneira que fizemos em `cellForRowAt` - Segunda mini-tarefa realizada!
+
+A terceira mini-tarefa é fazer a nova tela aparecer. Você já viu que view controllers possuem uma propriedade opcional chamada `storyboard` que pode conter o storyboard que os inicializou ou estar nula em caso de ele não ter sido inicializado a partir de um storyboard. Da mesma forma existe uma outra propriedade nos view controllers com o nome de `navigationController` que conterá o navigation controller em que ele está inserido caso este seja o caso, ou nulo em caso contrário.
```

Comment:
storyboard que os inicializou ou estar nula em caso de ele não ter sido inicializado a partir de um storyboard. 
Suggestion: I think you should use nil instead nulo or nula like you use english words: view controller, storyboard...